src/esp/io/json.cpp

```diff
@@ -55,6 +57,76 @@ bool writeJsonToFile(const JsonDocument& document,
   return writeSuccess;
 }
 
+int loadJsonIntoConfiguration(const JsonGenericValue& jsonObj,
+                              const std::string& subGroupName,
+                              esp::core::Configuration& config) {
+  // count number of valid user config settings found
+
+  int numConfigSettings = 0;
+  for (rapidjson::Value::ConstMemberIterator it = jsonObj.MemberBegin();
+       it != jsonObj.MemberEnd(); ++it) {
+    // for each key, attempt to parse
+    const std::string key = it->name.GetString();
+    const auto& obj = it->value;
+    // increment, assuming is valid object
+    ++numConfigSettings;
+
+    if (obj.IsFloat()) {
+      config.setSubgroupValue<float>(subGroupName, key, obj.GetFloat());
+    } else if (obj.IsDouble()) {
+      config.setSubgroupValue<double>(subGroupName, key, obj.GetDouble());
+    } else if (obj.IsNumber()) {
+      config.setSubgroupValue<int>(subGroupName, key, obj.GetInt());
+    } else if (obj.IsString()) {
+      config.setSubgroupValue<std::string>(subGroupName, key, obj.GetString());
+    } else if (obj.IsBool()) {
+      config.setSubgroupValue<bool>(subGroupName, key, obj.GetBool());
+    } else if (obj.IsArray() && obj.Size() > 0 && obj[0].IsNumber()) {
```

Comment:
Does this support obj.empty()? Might be cleaner here.

Reply:
Not sure what you mean. Could you elaborate? :)

Reply:
Actually, nevermind, this is rapidjson so it doesn't follow the STL.